### DISCUSSION TITLE
Is `lib/collection.js` unused?

### DISCUSSION THREAD

**Discussion by hasezoey** [Automattic/mongoose#12138] [Q&A] [answered]
I just noticed that the file `lib/collection.js` does not seem to be referenced anywhere, if this is true, should this maybe be removed?

PS: i also just noticed that it seems like `lib/promise_provider.js` is also unreferenced (could not find any import)

**Comment by vkarpov15** [comment, marked as answer]
`lib/collection.js` is used by `lib/drivers/node-mongodb-native/collection.js`. `Collection` is a base class that drivers inherit, like how [in-memory-driver does](https://github.com/mongoosejs/in-memory-driver/blob/501230102224eed198b2326f30e053869aedc199/src/collection.js#L4).

PromiseProvider is also used, for example here: https://github.com/Automattic/mongoose/blob/8f73e063531031ee4b889c8880e00765c452763b/lib/model.js#L1302

**Reply by hasezoey** [reply]
thanks, weird i didnt find these references through my editor (vscode), maybe i just not searched correctly
